code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm

```diff
@@ -2,45 +2,94 @@
 #define DEFILER_HEMODILE "Hemodile"
 #define DEFILER_TRANSVITOX "Transvitox"
 
+GLOBAL_LIST_INIT(defile_purge_list, typecacheof(list(
+	/datum/reagent/toxin/xeno_hemodile, /datum/reagent/toxin/xeno_transvitox, /datum/reagent/toxin/xeno_neurotoxin)))
+
 // ***************************************
-// *********** Sting
+// *********** Defile
 // ***************************************
-/datum/action/xeno_action/activable/larval_growth_sting/defiler
+/datum/action/xeno_action/activable/defile
 	name = "Defile"
 	action_icon_state = "defiler_sting"
-	mechanics_text = "Channel to inject an adjacent target with larval growth serum. At the end of the channel your target will be infected."
+	mechanics_text = "Channel to inject an adjacent target with an accelerant that violently reacts with xeno toxins, releasing gas and dealing heavy tox damage in proportion to the amount in their system."
 	ability_name = "defiler sting"
-	plasma_cost = 150
-	cooldown_timer = 20 SECONDS
+	plasma_cost = 100
+	cooldown_timer = 15 SECONDS
 	target_flags = XABB_MOB_TARGET
+	keybind_signal = COMSIG_XENOABILITY_DEFILE
 
-/datum/action/xeno_action/activable/larval_growth_sting/defiler/on_cooldown_finish()
+/datum/action/xeno_action/activable/defile/on_cooldown_finish()
 	playsound(owner.loc, 'sound/voice/alien_drool1.ogg', 50, 1)
-	to_chat(owner, span_xenodanger("You feel your toxin glands refill, another young one ready for implantation. You can use Defile again."))
+	to_chat(owner, span_xenodanger("You feel your toxin accelerant glands refill. You can use Defile again."))
 	return ..()
 
-/datum/action/xeno_action/activable/larval_growth_sting/defiler/use_ability(atom/A)
+/datum/action/xeno_action/activable/defile/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+
+//	var/mob/living/carbon/victim = A
+
+	if(!A.can_sting())
+		if(!silent)
+			to_chat(owner, span_xenodanger("Our sting won't affect this target!"))
+		return FALSE
+
+	if(!owner.Adjacent(A))
+		var/mob/living/carbon/xenomorph/X = owner
+		if(!silent)
+			to_chat(X, span_warning("We can't reach this target! We need to be adjacent!"))
+		return FALSE
+
+
+/datum/action/xeno_action/activable/defile/use_ability(atom/A)
 	var/mob/living/carbon/xenomorph/Defiler/X = owner
```

Comment:
at any point, do you need it to be a defiler? If not, just cast as a xeno, this allows action to be portable